teos/logger.py

```diff
@@ -14,20 +15,52 @@
 
 configured = False  # set to True once setup_logging is called
 
+timestamper = structlog.processors.TimeStamper(fmt="%d/%m/%Y %H:%M:%S")
 
-class JsonMsgLogger(logging.Logger):
+
+class FormattedSocketHandler(SocketHandler):
+    """ Works in the same way as SocketHandler, but it uses formatters. """
+
+    def emit(self, record):
+        """
+        Works exactly like SocketHandler.emit but formats the record before sending it.
+        record.args is set to none since they are already used by self.format, otherwise makePickle would try to
+        use them again and fail.
+
+        Args:
+            record (:obj:LogRecord <logging.LogRecord>): the record to be emitted.
+        """
+        try:
+            record.msg = self.format(record)
+            record.args = None
+
+            s = self.makePickle(record)
+            self.send(s)
+
+        except Exception:
+            self.handleError(record)
+
+
+class MultiNameFilter(logging.Filter):
     """
-    Works exactly like ``logging.Logger`` but represents dict messages as json. Useful to prevent dicts being cast
-    to strings via ``str()``.
+    Filters log records based on a list of logger names. Only loggers in the list are accepted.
     """
 
-    def makeRecord(self, *args, **kwargs):
-        """Makes a record where the message is json encoded."""
-        rv = super().makeRecord(*args, **kwargs)
-        if isinstance(rv.msg, dict):
-            rv.msg = json.dumps(rv.msg)
+    def __init__(self, names):
+        super().__init__()
+        self.names = names
 
-        return rv
+    def filter(self, record):
+        return record.name in self.names
+
+
+def add_component(logger, name, event_dict):
+    """
+    Adds the component to the entry if the entry is not from structlog. This only applies to the API since it is
+    handled by external WSGI servers.
+    """
+    event_dict["component"] = "API"
+    return event_dict
```

Comment:
I think it would be better to also specialized the name of the method to `add_api_component`, as the current name suggests a more generic method.

Reply:
Changed to `add_api_component`